scripts/dts/edtlib.py

```diff
@@ -1206,6 +1239,37 @@ def __repr__(self):
         return "<ControllerAndData, {}>".format(", ".join(fields))
 
 
+class PinCtrl:
+    """
+    Represents a single pin configuration on a device, e.g. pinctrl-0 or
+    pinctrl-1.
+
+    These attributes are available on PinCtrl objects:
+
+    node:
+      The Node instance the pinctrl-* property is on
+
+    name:
+      The name of the configuration, as given in pinctrl-names, or None if
+      there is no pinctrl-names property
```

Comment:
If we do not have an entry in pinctrl-names the property name (aka 'pinctrl-x') would be handy as a name.

Reply:
Maybe a `PinCtrl.index` field could be added instead. Not sure how useful it would be though, since the order in `Node.pinctrls` already gives the index, so maybe it could be left out until a need comes up.

Reply:
As long as the order is guaranteed to follow the pinctrl index it should be sufficient.